docs/advanced/additional-requests.rst

```diff
@@ -919,3 +889,118 @@ an :class:`~.HttpClient` instance:
+like the ones above, then it would cause the code to look like:
+
+.. code-block:: python
+
+    import attrs
+    import web_poet
+
+    import aiohttp
+    import requests
+    import urllib
+
+
+    @attrs.define
+    class SomePage(web_poet.ItemWebPage):
+        http_client: web_poet.HttpClient
+
+        async def to_item(self):
+            try:
+                response = await self.http_client.get("...")
+            except (aiohttp.ClientError, requests.RequestException, urllib.error.HTTPError):
+                # handle the error here
+
+Such code could turn messy in no time especially when the number of HTTP backends
+that Page Objects **should support** are steadily increasing. Not to mention the 
+plethora of exception types that HTTP libraries have. This means that Page
+Objects aren't truly portable in different types of frameworks or environments.
+Rather, they're only limited to work in the specific framework they're supported.
+
+In order for Page Objects to easily work in different Downloader Implementations,
+the framework that implements the HTTP Downloader backend should be able to raise
+exceptions from the :mod:`web_poet.exceptions.http` module in lieu of the backend
+specific ones `(e.g. aiohttp, requests, urllib, etc.)`.
+
+This makes the code much simpler:
+
+.. code-block:: python
+
+    import attrs
+    import web_poet
+
+
+    @attrs.define
+    class SomePage(web_poet.ItemWebPage):
+        http_client: web_poet.HttpClient
+
+        async def to_item(self):
+            try:
+                response = await self.http_client.get("...")
+            except web_poet.exceptions.HttpError:
+                # handle the error here
+
+Expected behavior for Exceptions
+********************************
+
+All exceptions that the HTTP Downloader Implementation (see :ref:`advanced-downloader-impl`
+doc section) explicitly raises when implementing it for **web-poet** should be
+:class:`web_poet.exceptions.http.HttpError` *(or a subclass from it)*. 
+
+For frameworks that implement and use **web-poet**, exceptions that ocurred when
+handling the additional requests like `connection errors`, `TLS errors`, etc should
+be replaced by :class:`web_poet.exceptions.http.HttpRequestError` by raising it
+explicitly.
+
+For responses that are not really errors like in the ``100-3xx`` status code range,
+no exception should be raised at all. For responses with status codes in 
+the ``400-5xx`` range, **web-poet** raises the :class:`web_poet.exceptions.http.HttpResponseError`
+exception.
+
+From this distinction, the framework shouldn't raise :class:`web_poet.exceptions.http.HttpResponseError`
+on its own at all, since the :class:`~.HttpClient` already handles that. However,
+the implementing framework could modify which status codes to allow (*meaning no
+exceptions raised*) using the ``allow_status`` parameter in the
+:class:`~.HttpClient` methods.
```

Comment:
I'm not sure I understand this part. Shouldn't it be like this:

1. Framework never raises HttpResponseError. Framework also has nothing to do with allow_status, it doesn't need to know about it.
2. web-poet takes a response, and can raise HttpResponseError, depending on allow_status value.

Reply:
I've corrected this in https://github.com/scrapinghub/web-poet/pull/38/commits/c5244b9fe04201529f3d78733e7cdeffcf7a4d54. Thanks!